streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java

```diff
@@ -87,25 +97,171 @@ public static void main(String[] args) throws Exception {
         System.out.println("kafka=" + kafka);
         System.out.println("zookeeper=" + zookeeper);
         System.out.println("stateDir=" + stateDir);
+        System.out.println("numRecords=" + numRecords);
 
         SimpleBenchmark benchmark = new SimpleBenchmark(stateDir, kafka, zookeeper);
 
         // producer performance
-        benchmark.produce();
+        benchmark.produce(SOURCE_TOPIC, VALUE_SIZE, "simple-benchmark-produce", numRecords, true, numRecords, true);
         // consumer performance
-        benchmark.consume();
+        benchmark.consume(SOURCE_TOPIC);
         // simple stream performance source->process
-        benchmark.processStream();
+        benchmark.processStream(SOURCE_TOPIC);
         // simple stream performance source->sink
-        benchmark.processStreamWithSink();
+        benchmark.processStreamWithSink(SOURCE_TOPIC);
         // simple stream performance source->store
-        benchmark.processStreamWithStateStore();
+        benchmark.processStreamWithStateStore(SOURCE_TOPIC);
+        // simple streams performance KSTREAM-KTABLE join
+        benchmark.kStreamKTableJoin(JOIN_TOPIC_1_PREFIX + "kStreamKTable", JOIN_TOPIC_2_PREFIX + "kStreamKTable");
+        // simple streams performance KSTREAM-KSTREAM join
+        benchmark.kStreamKStreamJoin(JOIN_TOPIC_1_PREFIX + "kStreamKStream", JOIN_TOPIC_2_PREFIX + "kStreamKStream");
+        // simple streams performance KTABLE-KTABLE join
+        benchmark.kTableKTableJoin(JOIN_TOPIC_1_PREFIX + "kTableKTable", JOIN_TOPIC_2_PREFIX + "kTableKTable");
+    }
+
+    /**
+     * Measure the performance of a KStream-KTable left join. The setup is such that each
+     * KStream record joins to exactly one element in the KTable
+     */
+    public void kStreamKTableJoin(String kStreamTopic, String kTableTopic) throws Exception {
+        CountDownLatch latch = new CountDownLatch(1);
+        int valueSizeBytes = VALUE_SIZE;
+
+        // initialize topics
+        System.out.println("Initializing kStreamTopic " + kStreamTopic);
+        produce(kStreamTopic, valueSizeBytes, "simple-benchmark-produce-kstream", numRecords, false, numRecords, false);
+        System.out.println("Initializing kTableTopic " + kTableTopic);
+        produce(kTableTopic, valueSizeBytes, "simple-benchmark-produce-ktable", numRecords, true, numRecords, false);
+
+        // perform join
+        final KafkaStreams streams = createKafkaStreamsKStreamKTableJoin(kStreamTopic, kTableTopic, numRecords, stateDir, kafka, zookeeper, latch);
+
+        Thread thread = new Thread() {
+            public void run() {
+                streams.start();
+            }
+        };
+        thread.start();
+
+        long startTime = System.currentTimeMillis();
+
+        while (latch.getCount() > 0) {
+            try {
+                latch.await();
+            } catch (InterruptedException ex) {
+                Thread.interrupted();
```

Comment:
I'm not sure this is necessary here as the interrupt flag is not going to be checked anywhere else by this `Thread`